src/backend/gp_libpq_fe/fe-exec.c

```diff
@@ -2968,7 +2968,7 @@ PQoidStatus(const PGresult *res)
 
 	size_t		len;
 
-	if (!res || !res->cmdStatus || strncmp(res->cmdStatus, "INSERT ", 7) != 0)
+	if (!res || strncmp(res->cmdStatus, "INSERT ", 7) != 0)
```

Comment:
I do not think assuming strncpy(NULL, ...) is ok is a good idea. At least I quickly searched on stackoverflow,

http://stackoverflow.com/questions/21865041/in-c-what-exactly-happens-when-you-pass-a-null-pointer-to-strcmp

Although this page is probably not accurate, but your change might risk not working in some lib/platform implementation.